### PR TITLE
nixosTests.docker-containers: Port to python

### PR DESCRIPTION
**Pull request by tfc**
###### Motivation for this change

#72828

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tfc** [comment]
@GrahamcOfBorg test docker-containers

File: nixos/tests/docker-containers.nix [hunk 1]
```diff
@@ -1,30 +1,27 @@
...
 
   testScript = ''
-    startAll;
-    $docker->waitForUnit("docker-nginx.service");
-    $docker->waitForOpenPort(8181);
-    $docker->waitUntilSucceeds("curl http://localhost:8181|grep Hello");
+    start_all()
+    docker.wait_for_unit("docker-nginx.service")
+    docker.wait_for_open_port(8181)
+    docker.wait_until_succeeds("curl http://localhost:8181 | grep Hello")
```
Comment (worldofpeace) [inline nixos/tests/docker-containers.nix]: Can we not use `grep`?

Comment (tfc) [reply nixos/tests/docker-containers.nix]: what do you mean?

If something is a one-shot command and we want to see if it has certain content, then we do `assert X in machine.succeed("foo")`, but when we do `wait_until_succeeds` we don't have this option.

Comment (worldofpeace) [reply nixos/tests/docker-containers.nix]: Understood.

**Review by worldofpeace — APPROVED** [review]

**Pull request merged**